src/NodeConnectionInteraction.hpp

```diff
@@ -5,30 +5,34 @@
 #include "Node.hpp"
 #include "Connection.hpp"
 
+class DataModelRegistry;
+class FlowScene;
+
 /// Class performs various operations on the Node and Connection pair.
 /// An instance should be created on the stack and destroyed when
 /// the operation is completed
 class NodeConnectionInteraction
 {
 public:
   NodeConnectionInteraction(Node& node,
-                            Connection& connection)
-    : _node(&node)
-    , _connection(&connection)
-  {}
+                            Connection& connection,
+                            FlowScene& scene);
```

Comment:
I originally tried to do that way, but to create nodes and connections, I needed the scene, so I swapped the registry to it. (tryConnect 1.5) The entire code segment that that handles 1.5, could be extracted, and put into a class that has a reference to the scene, if you think thats a better solution.